From 51c857b238ce176fc35dcbbaa0abf4e2423eacf1 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:04:19 -0400
Subject: [kernel] perf: Fix perf_event_init_context()

Message-id: <1410181752-25631-15-git-send-email-jolsa@redhat.com>
Patchwork-id: 91691
O-Subject: [PATCH RHEL7.1 BZ1134356 014/307] perf: Fix perf_event_init_context()
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit ffb4ef21ac4308c2e738e6f83b6741bbc9b4fa3b
Author: Peter Zijlstra <peterz@infradead.org>
Date: Mon May 5 19:12:20 2014 +0200

description
===========
perf_pin_task_context() can return NULL but perf_event_init_context()
assumes it will not, correct this.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/kernel/events/core.c b/kernel/events/core.c
index a1fc9db..c45e3d5 100644
--- a/kernel/events/core.c
+++ b/kernel/events/core.c
@@ -7733,6 +7733,8 @@ int perf_event_init_context(struct task_struct *child, int ctxn)
   * swapped under us.
   */
  parent_ctx = perf_pin_task_context(parent, ctxn);
+ if (!parent_ctx)
+  return 0;
 
  /*
   * No need to check if parent_ctx != NULL here; since we saw
-- 
1.7.1